### ISSUE TITLE
Xiaomi Mi Smart Computer Monitor Light Bar 1S (MJGJD02YL) show color picker instead of color temperature slider

### ISSUE THREAD

**Issue by Walhalla360** [al-one/hass-xiaomi-miot#1043]
### Device model / 设备型号

yeelink.light.lamp22

### Component version / 插件版本

0.7.7

### HA core version / HA版本

2023.3.1

### Integrated mode / 集成方式

Token

### The problem / 问题详情

Today I tried to get the Xiaomi Mi Smart Computer Monitor Light Bar 1S (MJGJD02YL) working with Home Assistant and first found the other, already deprecated, component from you named: [hass-miio-yeelink](https://github.com/al-one/hass-miio-yeelink).

I set it up and after that I found this component here and set it up again with the "host/token" option. First I tried the auto mode, but with that I couldn't even switch the lamp on/off.

There is only one problem with the new component, it displays a color picker instead of the color temperature slider.

![image](https://user-images.githubusercontent.com/4402538/222800529-13d17ed3-f691-42fb-af51-377f74c4a297.png)

This is a problem, because this lamp has no color support and so the adjustment doesn't work.

While checking the entities, I found out that the new component is setting "color_mode" to "brightness" instead of "color_temp".

I "fixed" it with the following change in "customize.yaml", but perhaps you can change it directly in the setup code.

light.light.yeelink_lamp22_XXXX_light:
  color_mode: 'color_temp'

### Entity attributes / 实体属性

**light.yeelink_lamp22_XXXX_light**

min_color_temp_kelvin: 2702
max_color_temp_kelvin: 6493
min_mireds: 154
max_mireds: 370
effect_list: Reading, Office, Leisure, Warmth, Computer
supported_color_modes: brightness, color_temp, onoff
color_mode: brightness
brightness: 64
model: yeelink.light.lamp22
lan_ip: XXX.XXX.XXX.XXX
mac_address: XX:XX:XX:XX:XX:XX
entity_class: MiotLightEntity
miot_type: urn:miot-spec-v2:device:light:0000A001:yeelink-lamp22:1:0000C802
light.on: true
light.brightness: 25
light.color_temperature: 2703
light.default_power_on_state: true
light.mode: 0
yl_light.save_state: true
yl_light.delay_time: 15
yl_light.delay_off_enable: false
yl_light.double_click_enable: false
yl_light.double_click: 1
yl_light.long_press_enable: false
yl_light.scene_mode: 0
yl_light.long_press_one: 9
yl_light.long_press_two: 8
yl_light.long_press_three: 5
yl_light.work_minutes: 45
yl_light.break_minutes: 10
yl_light.pomodoro_enable: false
yl_light.mode_one_bright: 50
yl_light.mode_one_ct: 4000
yl_light.mode_two_bright: 100
yl_light.mode_two_ct: 2700
yl_light.mode_three_bright: 30
yl_light.mode_three_ct: 5000
yl_light.mode_four_bright: 60
yl_light.mode_four_ct: 3000
yl_light.ble_paired_mac: XX:XX:XX:XX:XX:XX
state_updater: lan
friendly_name: Xiaomi Miot light
supported_features: 36

**light.xiaomi_yeelink (old)**

min_color_temp_kelvin: 2702
max_color_temp_kelvin: 5714
min_mireds: 175
max_mireds: 370
supported_color_modes: color_temp
color_mode: color_temp
brightness: 64
color_temp_kelvin: 2710
color_temp: 369
hs_color: 28.373, 65.405
rgb_color: 255, 167, 88
xy_color: 0.524, 0.388
model: yeelink.light.lamp22
lan_ip: XXX.XXX.XXX.XXX
mac_address: XX:XX:XX:XX:XX:XX
firmware_version: 2.1.7_0020
hardware_version: esp32
entity_class: YeelightEntity
power: on
nl_br: 
delayoff: 0
bright: 25
ct: 2703
friendly_name: Xiaomi Yeelink
supported_features: 3

**Comment by al-one** [comment]
```yaml
supported_color_modes: brightness, color_temp, onoff
color_mode: brightness
```

It can be seen from the entity attributes that the current `color_mode` should be `brightness`, so I think the card display error is caused by a bug in the HA frontend.

**Comment by Walhalla360** [comment]
yeah, but who sets the `color_mode` to `brightness` is this coming from this component or HA? Because the correct setting should be `color_temp` if the device supports it or not?

**Comment by al-one** [comment]
> who sets the `color_mode` to `brightness` is this coming from this component or HA?

This component.


> Because the correct setting should be color_temp if the device supports it or not?

Yes.

**Comment by Walhalla360** [comment]
Ok, so is there a way for you to change it or do you need more input from me? Because when I change the `color_mode` to `color_temp`, like I mentioned, I have a borked color temperature slider:

![image](https://user-images.githubusercontent.com/4402538/222935003-11bc2202-e456-4b0c-a499-9b237cea15ed.png)

It is not displaying the label with the current color temperature and the current slider position, until I click somewhere on the bar and the adjustment of the color temperature is a bit "choppy". 

Or do you think that `color_mode: brightness` is not the issue that it is displaying the color picker instead of the color temperature slider, when I open the card and click on the "Change color" icon?

I mean, I have a brightness slider, when I click on the card, but "Change color" is broken and all of my other lamps (from other components) with color temperature feature have `color_mode: color_temp` set and the slider is working like a charm.

![image](https://user-images.githubusercontent.com/4402538/222935597-cd9395eb-e2ea-4a45-a9df-65e30ce5b3b6.png)

**Comment by misternikitos** [comment]
Hello everyone. I also faced the problem of incorrect display of the slider color temperature.
I checked all the light entities that have color temperature adjustment. All have `supported_color_modes: color_temp` (without brightness and onoff).
Using file editor, I removed `brightness` and `on off` from this attribute in the core.entity_registry file. After that, the entity began to work as it should.

**Comment by al-one** [comment]
Try master branch.

**Comment by Walhalla360** [comment]
Thank you very much for you effort, it is fully fixed for me now.
